sys/include/net/netopt.h

```diff
@@ -32,40 +32,66 @@ extern "C" {
 /**
  * @brief   Global list of configuration options available throughout the
  *          network stack, e.g. by netdev and netapi
+ *
+ * The data type specified in parentheses for each individual option is the
+ * data type to use for the argument when getting/setting the value of the option.
+ *
+ * All arguments longer than 1 byte (e.g. uint16_t) are given in host byte order
+ * unless anything else is specified below.
  */
 typedef enum {
-    NETOPT_CHANNEL,             /**< get/set channel as uint16_t in host
-                                 *   byte order */
-    NETOPT_IS_CHANNEL_CLR,      /**< check if channel is clear */
-    NETOPT_ADDRESS,             /**< get/set address in host byte order */
+    /**
+     * @brief   (uint16_t) channel number
+     */
+    NETOPT_CHANNEL,
+    /**
+     * @brief   (@ref netopt_enable_t) check whether the network medium is clear
+     *
+     * Getting this option can be used to trigger a manual clear channel
+     * assessment (CCA) on some wireless devices.
+     */
+    NETOPT_IS_CHANNEL_CLR,
+    /**
+     * @brief   (byte array, see below) link layer address
+     *
+     * Device type   | Length | Meaning
+     * ------------- | ------ | -----
+     * IEEE 802.15.4 | 2      | device short address
+     * Ethernet      | 6      | device MAC address
+     */
+    NETOPT_ADDRESS,
 
     /**
-     * @brief    get/set long address in host byte order
+     * @brief   (byte array, see below) long link layer address
```

Comment:
somewhere here you should mention that the address is to be in network byte order (see your comment above).